### PR TITLE
Fix image clients for non-batching models (like example onnx models)

### PR DESCRIPTION
**Pull request by CoderHam**
Fixes image model tests for onnx.

### CODE REVIEW

**Comment by CoderHam** [comment]
@deadeyegoodwin I saw your PR for the client library rework to use TritonJson. I have not incorporated those changes here yet.

File: src/clients/c++/examples/image_client.cc [hunk 1]
```diff
@@ -930,11 +940,7 @@ main(int argc, char** argv)
   std::vector<int64_t> shape;
   // Include the batch dimension if required
   if (model_info.max_batch_size_ != 0) {
-    if (batch_size > 0) {
```
Comment (deadeyegoodwin) [inline src/clients/c++/examples/image_client.cc:933]: There is a test above for batch_size > 0 so don't need this.

File: src/clients/c++/examples/image_client.cc [hunk 2]
```diff
@@ -211,12 +211,21 @@ Postprocess(
...
-  // Validate shape
-  if ((shape.size() != 2) || (shape[0] != (int)batch_size) ||
-      (shape[1] != (int)topk)) {
-    std::cerr << "received incorrect shapes for " << output_name << std::endl;
-    exit(1);
+
+  // Validate shape. Special handling for non-batch model
+  if (!batching) {
+    if ((shape.size() != 1) || (shape[0] != (int)topk)) {
+      std::cerr << "received incorrect shapes for " << output_name << std::endl;
```
Comment (deadeyegoodwin) [inline src/clients/c++/examples/image_client.cc:218]: "incorrect shape"

File: src/clients/c++/examples/image_client.cc [hunk 3]
```diff
@@ -211,12 +211,21 @@ Postprocess(
...
+      exit(1);
+    }
+  } else {
+    if ((shape.size() != 2) || (shape[0] != (int)batch_size) ||
+        (shape[1] != (int)topk)) {
+      std::cerr << "received incorrect shapes for " << output_name << std::endl;
```
Comment (deadeyegoodwin) [inline src/clients/c++/examples/image_client.cc:224]: "incorrect shape"

File: src/clients/c++/examples/image_client.cc (hunk 1 above)
Comment (CoderHam) [reply src/clients/c++/examples/image_client.cc:933]: Yes I removed this in this PR itself.

Comment (deadeyegoodwin) [reply src/clients/c++/examples/image_client.cc:933]: Ha, yes. I looked at it backwards.

**Pull request merged**